lib/src/http/http_body.dart

```diff
@@ -56,21 +56,63 @@ class HttpBody extends BaseHttpBody {
   Encoding _encoding;
 
   /// Construct the body to an HTTP request or an HTTP response from bytes.
+  ///
+  /// If [encoding] is given, it will be used to encode/decode the body.
+  ///
+  /// Otherwise, the `charset` parameter from [contentType] will be mapped to an
+  /// Encoding supported by Dart.
+  ///
+  /// If a `charset` parameter is not available or the value is unrecognized,
+  /// [fallbackEncoding] will be used.
+  ///
+  /// If [fallbackEncoding] is `null`, UTF8 will be the default encoding.
+  ///
+  /// If an encoding cannot be parsed from the content-type header (via the
+  /// `charset` param), then [fallbackEncoding] will be used (UTF8 by default).
   HttpBody.fromBytes(MediaType this.contentType, List<int> bytes,
-      {Encoding fallbackEncoding}) {
-    _encoding = http_utils.parseEncodingFromContentType(contentType,
-        fallback: fallbackEncoding);
+      {Encoding encoding, Encoding fallbackEncoding}) {
+    if (fallbackEncoding == null) {
+      fallbackEncoding = UTF8;
+    }
+    if (encoding != null) {
+      _encoding = encoding;
+    } else {
+      _encoding = http_utils.parseEncodingFromContentType(contentType,
+          fallback: fallbackEncoding);
+    }
+
     if (bytes == null) {
       bytes = [];
     }
     _bytes = new Uint8List.fromList(bytes);
   }
 
   /// Construct the body to an HTTP request or an HTTP response from text.
+  ///
+  /// If [encoding] is given, it will be used to encode/decode the body.
+  ///
+  /// Otherwise, the `charset` parameter from [contentType] will be mapped to an
+  /// Encoding supported by Dart.
+  ///
+  /// If a `charset` parameter is not available or the value is unrecognized,
+  /// [fallbackEncoding] will be used.
+  ///
+  /// If [fallbackEncoding] is `null`, UTF8 will be the default encoding.
+  ///
+  /// If an encoding cannot be parsed from the content-type header (via the
+  /// `charset` param), then [fallbackEncoding] will be used (UTF8 by default).
   HttpBody.fromString(MediaType this.contentType, String body,
-      {Encoding fallbackEncoding}) {
-    _encoding = http_utils.parseEncodingFromContentType(contentType,
-        fallback: fallbackEncoding);
+      {Encoding encoding, Encoding fallbackEncoding: UTF8}) {
```

Comment:
Good catch, I had it this way at first but switched to Max's suggestion of just populating `fallbackEncoding` to UTF8 if it's null instead of throwing an ArgumentError. I'll fix this